### ISSUE TITLE
[TypeScript target] Missing Recognizer.getErrorListenerDispatch()

### ISSUE THREAD

**Issue by kaby76** [antlr/antlr4#4437]
`Recognizer.getErrorListenerDispatch()` is missing from the TypeScript target. That function is needed to call `this.getErrorListenerDispatch().syntaxError()` from the lexer, in a base class. For an example, [see this code in the lua grammar](https://github.com/antlr/grammars-v4/blob/4d3b08fa77c5820a1cdeaf6a016aee7ce1c7fdeb/lua/Java/LuaLexerBase.java#L46-L47). It's available in all other targets.

https://github.com/antlr/antlr4/blob/487cb28bd359587e67794b25b144b7df83ddf1a2/runtime/JavaScript/src/antlr4/Recognizer.js#L129-L131

**Comment by ericvergnaud** [comment]
Correct. I think it's also missing ProxyErrorListener. Can you provide a PR ?

**Comment by kaby76** [comment]
Will do later today!
